driver/controller.go

```diff
@@ -181,20 +181,15 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		volumeReq.SnapshotID = snapshotID
 	}
 
-	log.Info("checking volume limit")
-	details, err := d.checkLimit(ctx)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to check volume limit: %s", err)
-	}
-	if details != nil {
-		return nil, status.Errorf(codes.ResourceExhausted,
-			"volume limit (%d) has been reached. Current number of volumes: %d. Please contact support.",
-			details.limit, details.numVolumes)
-	}
-
 	log.WithField("volume_req", volumeReq).Info("creating volume")
-	vol, _, err := d.storage.CreateVolume(ctx, volumeReq)
+	vol, cvResp, err := d.storage.CreateVolume(ctx, volumeReq)
 	if err != nil {
+		var errorResponse *godo.ErrorResponse
+		if errors.As(err, &errorResponse) && strings.Contains(err.(*godo.ErrorResponse).Message, "capacity limit exceeded") &&
```

Comment:
I just realized that we do not need to check for and cast into an `ErrorResponse` now that we have discovered that the error ID isn't exported and that we can only check for a portion of the error message: the [`Error()` implementation](https://github.com/digitalocean/godo/blob/2e7120f15645a6399cb37a949bf9a6168b4b6212/godo.go#L476) always includes it, which means it suffices to check `err.Error()` for it.

Reply:
True, I refactored the code to check the substring from the basic `err.Error()`.